spk/pyload/src/installer.sh

```diff
@@ -0,0 +1,146 @@
+            chmod 775 "/${VOLUME}/${WIZ_SHARE}"
+            echo "WIZ_SHARE=${WIZ_SHARE}" > "${SYNOPKG_PKGINST_TEMP_DIR}/etc/pyload-pkg.conf"
+        else
+            if [ -z "$(echo "${SHARE_EXISTS}" | grep "${VOLUME}")" ]; then
+               # IDEA: Shall I rewrite ${wizard_download_dir} to existing share name on different volume or end with error? For now, end with error.
+               echo "ERROR: There's already a share '${WIZ_SHARE}' but it points to a different volume." > ${SYNOPKG_TEMP_LOGFILE}
+               exit 1
+            fi
+        fi
+
+        # At this state, ${wizard_download_dir} is in a shared folder. Regardless of ${wizard_download_dir} existence.
+
+        if [ ! -d ${wizard_download_dir} ]; then
+            mkdir -p "${wizard_download_dir}" && \
+                chown admin:users "${wizard_download_dir}" && \
+                chmod 775 "${wizard_download_dir}"
+            echo "WIZ_DWNLD_DIR=${wizard_download_dir}" >> "${SYNOPKG_PKGINST_TEMP_DIR}/etc/pyload-pkg.conf"
+        fi
+    fi
+    exit 0
+}
+
+postinst ()
+{
+    # Link
+    ln -s ${SYNOPKG_PKGDEST} ${INSTALL_DIR}
+
+    # Set cfg location
+    echo "${INSTALL_DIR}/etc" > "${INSTALL_DIR}/share/pyload/module/config/configdir"
+
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        test "${wizard_download_dir}" || wizard_download_dir="/volume1/downloads"
+
+        # Setting cfg
+        sed -e " \
+            s|@W_DOWNLOAD_DIR@|${wizard_download_dir}|; \
+            s|@LNG@|$(lng2iso ${SYNOPKG_DSM_LANGUAGE})| \
+               " \
+                  < "${INSTALL_DIR}/share/pyload/module/config/default_syn.conf" \
+                  > "${INSTALL_DIR}/etc/pyload.conf"
+
+        # hash password
+        SALT=$((RANDOM%99999+10000))
+        SALTED_PW_HASH=${SALT}$(echo -n "${SALT}${wizard_password}" | openssl dgst -sha1 2>/dev/null | cut -d" " -f2)
+
+        # init DB & add 'admin' user
+        echo -n "4" > "${INSTALL_DIR}/etc/files.version"
+        sqlite3 "${INSTALL_DIR}/etc/files.db" < "${INSTALL_DIR}/share/pyload/module/config/pyload_init.sql" > ${SYNOPKG_TEMP_LOGFILE} || exit 1
+        sqlite3 "${INSTALL_DIR}/etc/files.db" "INSERT INTO users (name, password) VALUES ('admin', '${SALTED_PW_HASH}')" > ${SYNOPKG_TEMP_LOGFILE} || exit 1
+
+        # Additional (optional) config/examples
+        cp -f "${INSTALL_DIR}/share/pyload/module/config/htaccess.example" "${INSTALL_DIR}/etc/"
+    fi
+    
+    # Add firewall config
+    ${SERVICETOOL} --install-configure-file --package ${FWPORTS} >> /dev/null
+
+    exit 0
+}
+
+preuninst ()
+{
+    if [ "${SYNOPKG_PKG_STATUS}" == "UNINSTALL" ]; then
+	test -f "${INSTALL_DIR}/etc/pyload-pkg.conf" && . "${INSTALL_DIR}/etc/pyload-pkg.conf"
+        if [ "${WIZ_DWNLD_DIR}" ]; then
+            :   # IDEA: If download folder was created by postinst(), remove it. But do I know whether user didn't start using it for anything else after install?
```

Comment:
To answer the question: We never remove dirs. Once created (regardless of how that happened), they are not yours to control anymore.